contracts/external/policies/ERC20SpendingLimitPolicy.sol

```diff
@@ -149,4 +120,52 @@ contract ERC20SpendingLimitPolicy is IActionPolicy {
             return VALIDATION_SUCCESS;
         }
     }
+
+    /**
+     * @notice Supports the IERC165 interface.
+     * @param interfaceID The interface ID.
+     * @return True if the interface is supported, false otherwise.
+     */
+    function supportsInterface(bytes4 interfaceID) external pure override returns (bool) {
+        return (
+            interfaceID == type(IERC165).interfaceId || interfaceID == type(IPolicy).interfaceId
+                || interfaceID == type(IActionPolicy).interfaceId
+        );
+    }
+
+    function _isTokenTransfer(
+        address account,
+        bytes calldata callData
+    )
+        internal
+        pure
+        returns (bool isTransfer, uint256 amount)
+    {
+        bytes4 functionSelector = bytes4(callData[0:4]);
+
+        if (functionSelector == IERC20.approve.selector) {
+            (, amount) = abi.decode(callData[4:], (address, uint256));
+            return (true, amount);
+        } else if (functionSelector == IERC20.transfer.selector) {
+            (, amount) = abi.decode(callData[4:], (address, uint256));
+            return (true, amount);
+        } else if (functionSelector == IERC20.transferFrom.selector) {
```

Comment:
do we need to check that from is account? could there be a case where someone tries to claim approved erc20s and is denied by this?

Reply:
good point - I think this makes sense

Reply:
fixed in https://github.com/erc7579/smartsessions/pull/122/commits/c728cbd4f2d216a1c54ba79699a5df093301e01d